libres/lib/include/ert/res_util/memory.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+   Copyright (C) 2011  Equinor ASA, Norway.
+
+   The file 'memory.hpp' is part of ERT - Ensemble based Reservoir Tool.
+
+   ERT is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   ERT is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or
+   FITNESS FOR A PARTICULAR PURPOSE.
+
+   See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
+   for more details.
+*/
+
+#include <cstddef>
+
+#include <ert/python.hpp>
+
+#ifndef ERT_MEMORY_H
+#define ERT_MEMORY_H
+
+namespace memory {
+
+std::size_t system_ram_free(void);
+std::size_t process_memory(void);
+std::size_t process_max_memory(void);
+std::size_t process_max_rss(void);
+
+class scoped_memory_logger {
+public:
+    scoped_memory_logger(std::shared_ptr<ert::ILogger> logger,
+                         const std::string &message)
+        : m_logger(logger), m_message(message), m_enter_mem(process_memory()),
+          m_enter_max_mem(process_max_memory()) {
+        m_logger->info("Enter {} Mem: {}Mb  Maxmem: {}Mb", m_message,
+                       m_enter_mem, m_enter_max_mem);
+    };
+
+    ~scoped_memory_logger() {
+        std::size_t cur_mem = process_memory();
+        std::size_t delta_cur_mem = cur_mem - m_enter_mem;
+        std::size_t max_mem = process_max_memory();
+        std::size_t delta_max_mem = max_mem - m_enter_max_mem;
+
+        m_logger->info(
+            "Exit {} Mem: {}Mb ({}{}) MaxMem: {}Mb ({}{})", m_message, cur_mem,
+            delta_cur_mem < 0 ? "-" : "+", cur_mem - m_enter_mem, max_mem,
```

Comment:
Ditto +2. I would move `cur_mem - m_enter_mem` outside of this call or explicitly mark them as `static_cast<ssize_t>(cur_mem - m_enter_mem)`.

Reply:
Mmm...  just realized that size_t may show some funky behaviour when subtracting two of them. I'll redo this logic...